src/components/Accordion.tsx

```diff
@@ -0,0 +1,185 @@
+      accordionToggleRef.current.dispatchEvent(tdsToggleEvent);
+    }
+  };
+
+  return (
+    <div>
+      <div className="tds-headline-02 tds-u-pb1">Accordion</div>
+      <tds-toggle ref={toggleRef}>
+        <div slot="label">{paddingReset ? 'Disable Padding Reset' : 'Enable Padding Reset'}</div>
+      </tds-toggle>
+      <br />
+      <tds-toggle ref={accordionToggleRef} checked={checkboxState} onTdsToggle={handleAccordionToggleChange}>
+        <div slot="label">{checkboxState ? 'Disable Accordion' : 'Enable Accordion'}</div>
+      </tds-toggle>
+      <h2>Icon Position</h2>
+      <div>
+      <label>
+          <input
+            type="radio"
+            name="icon-position"
+            value="end"
+            checked={iconPosition === 'end'}
+            onChange={handleIconPositionChange}
+          />
+          End
+        </label>
+        <label>
+          <input
+            type="radio"
+            name="icon-position"
+            value="start"
+            checked={iconPosition === 'start'}
+            onChange={handleIconPositionChange}
+          />
+          Start
+        </label>
+      </div>
+
+      <h2>Accordion with Primary Mode</h2>
+
+      <tds-accordion mode-variant="primary">
+        <tds-accordion-item
+          header="Item 1"
+          expanded={expandedItem === 'item1'}
+          expand-icon-position={iconPosition}
+          disabled={!checkboxState}
+          padding-reset={paddingReset}
+          onTdsToggle={() => handleAccordionItemClick('item1')}
+        >
+          This is the panel, which contains associated information with the header. Usually, it contains text, set in the same size as the header. Lorem ipsum doler sit amet.
+        </tds-accordion-item>
+        <tds-accordion-item
+          header="Item 2"
+          expanded={expandedItem === 'item2'}
+          expand-icon-position={iconPosition}
+          disabled={!checkboxState}
+          padding-reset={paddingReset}
+          onTdsToggle={() => handleAccordionItemClick('item2')}
+        >
+          This is the panel, which contains associated information with the header. Usually, it contains text, set in the same size as the header. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis laoreet vestibulum fermentum.
+        </tds-accordion-item>
+        <tds-accordion-item
+          header="Item 3"
+          expanded={expandedItem === 'item3'}
+          expand-icon-position={iconPosition}
+          disabled={!checkboxState}
+          padding-reset={paddingReset}
+          onTdsToggle={() => handleAccordionItemClick('item3')}
+        >
+          This is the panel, which contains associated information with the header. Usually, it contains text, set in the same size as the header. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis laoreet vestibulum fermentum.
+        </tds-accordion-item>
+      </tds-accordion>
+
+      <h2>Accordion with Secondary Mode</h2>
+
```

Comment:
Do I understand correctly that both accordions items right now are being opened/closed with the handleAccordionItemClick function. If so, I think it makes sense to keep one of them just using the accordions own functionality for opeing/closing.